langlib/lang.array/src/main/ballerina/src/lang.array/array.bal

```diff
@@ -48,50 +66,173 @@ public function iterator(Type[] arr) returns abstract object {
     return arrIterator;
 }
 
+# Returns a new array comprising of position and member pairs.
+#
+# + arr - the array
+# + return - array of position, member pair
 public function enumerate(Type[] arr) returns [int, Type][] = external;
 
-// Functional iteration
-
+# Returns a new array applying function `func` to each member of array `arr`.
+#
+# + arr - the array
+# + func - a function to apply to each member
+# + return - new array containing result of applying function `func` to each member
 public function 'map(Type[] arr, function(Type val) returns Type1 func) returns Type1[] = external;
 
+# Apply function `func` to each member of array `arr`.
+#
+# + arr - the array
+# + func - a function to apply to each member
 public function forEach(Type[] arr, function(Type val) returns () func) returns () = external;
 
+# Returns a new array constructed from those elements of 'arr' for which `func` returns true.
+#
+# + arr - the array
+# + func - a predicate to apply to each element to determine if it should be included
+# + return - new array only containig members which evaluate function 'func' to true
 public function filter(Type[] arr, function(Type val) returns boolean func) returns Type[] = external;
 
+# Reduce operate on each member of `arr` using combining function `func` to produce
+# a new value combining all members of `arr`.
+#
+# + arr - the array
+# + func - combining function
+# + initial - initial value to first evaluation of combining function `func`
+# + return - result of applying combining function to each member of the array
+#
+# Example: Emulating sum function.
+# ```
+# var ar = [1, 2, 3];
+# var a = ar.reduce(function (int i, int j) returns int { return i + j; }, 0);
+# ```
+#
+# Example: Emulating map behavior.
+# ```
+# var ar = [1, 2, 3];
+# int[] newArr = [];
+# int[] a = ar.reduce(function (int[] a, int j) returns int[] { a.push(j*2); return a; }, newArr);
+# ```
 public function reduce(Type[] arr, function(Type1 accum, Type val) returns Type1 func, Type1 initial) returns Type1 = external;
 
-// Sub-array
-
-
-// TODO Fix this.
+// TODO: Add default arg for `endIndex`
+# Returns a sub array starting from `startIndex` (inclusive) to `endIndex` (exclusive).
+#
+# + arr - the array
+# + startIndex - index of first member to include in the slice
+# + endIndex - index of first member not to include in the slice
+# + return - array slice within specified range
 public function slice(Type[] arr, int startIndex, int endIndex) returns Type[] = external;
 
+# Removes the member of `arr` and index `i` and returns it.
```

Comment:
```suggestion
# Removes the member of `arr` and index `i`, and returns it.
```
Did we mean both the member of `arr` and index `i` are removed? If so, we need this comma.

Reply:
'and' should be 'at'.
Removes the member of `arr` at index `i`, and return the removed member.